dali/kernels/tensor_view.h

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_TENSOR_VIEW_H_
+#define DALI_KERNELS_TENSOR_VIEW_H_
+
+#include "shape.h"
+
+namespace tensor {
+
+struct EmptyBackendTag {};
+
+template <typename Backend, typename DataType, int ndim>
+struct TensorView;
+
+template <typename Backend, typename DataType>
+struct TensorView<Backend, DataType, DynamicDimensions> {
+  TensorView() : data(nullptr), shape() {}
```

Comment:
Why not defaulted constructor + member initialization for `data`? Member initialization is more robust against tinkering with constructors.